x/finality/keeper/liveness.go

```diff
@@ -200,3 +182,21 @@ func (k Keeper) updateSigningInfo(
 
 	return modifiedSignInfo, &signInfo, nil
 }
+
+func (k Keeper) jailSluggishFinalityProvider(ctx context.Context, fpBtcPk *types.BIP340PubKey) error {
+	err := k.hooks.AfterSluggishFinalityProviderDetected(ctx, fpBtcPk)
+	if err != nil {
+		return err
+	}
```

Comment:
How about we call `k.BTCStakingKeeper.JailFinalityProvider` directly? From what I see the `AfterSluggishFinalityProviderDetected` hook is only used here and its only functionality is to call `JailFinalityProvider`. Removing this hook relationship makes the code easier to understand

Reply:
We would have `slashing reward` logic in the future. So better to insert it here?

Reply:
We can add the logic of slashing rewards into `JailFinalityProvider`, as removing FP voting power and slashing its reward always need to be done at the same time. No strong opinion on this though

Reply:
Agreed that jailing and slashing rewards should happen at the same time, so we can achieve this by having both modules implement the same hook. Some [discussion](https://github.com/babylonchain/babylon/pull/681#discussion_r1660620692) about why we decided to use a hook.

Reply:
Sure. In fact `x/finality` module has access to `x/incentive` so it can invoke the incentive module to slash the reward. But no problem if you think this is better done in a hook